central/sensor/service/pipeline/imageintegrations/pipeline.go

```diff
@@ -124,7 +124,7 @@ func matchesECRAuth(new, old *storage.ImageIntegration) bool {
 	return newAuth == nil && oldAuth == nil ||
 		newAuth.GetUsername() == oldAuth.GetUsername() &&
 			newAuth.GetPassword() == oldAuth.GetPassword() &&
-			newAuth.GetExpiresAt().Equal(oldAuth.GetExpiresAt())
+			newAuth.GetExpiresAt().AsTime().Equal(oldAuth.GetExpiresAt().AsTime())
```

Comment:
is `GetExpiresAt()` nil safe?

Reply:
If `newAuth` is nil, `GetExpiresAt()` will return `nil`. So it is `nil` safe. And `AsTime()` will consider `nil` as `0` timestamp.